msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -706,12 +674,14 @@ public void acquireTokenAsync(@NonNull final AcquireTokenParameters acquireToken
     }
 
     protected AccountRecord getAccountRecord(@Nullable final IAccount account) {
-        if (account != null) {
+        final MultiTenantAccount multiTenantAccount = (MultiTenantAccount) account;
+
+        if (null != multiTenantAccount && null != multiTenantAccount.getHomeAccountId()) {
             return AccountAdapter.getAccountInternal(
                     mPublicClientConfiguration.getClientId(),
                     mPublicClientConfiguration.getOAuth2TokenCache(),
-                    account.getHomeAccountIdentifier().getIdentifier(),
-                    AccountAdapter.getRealm(account)
+                    multiTenantAccount.getHomeAccountId(),
+                    multiTenantAccount.getTenantId() // TODO This assumes the home account, but what about guests??
```

Comment:
TODO?

Reply:
@shoatman I'll resolve this in the morning (6/5/19) -- feel free to review the rest and I'll submit this fix before merge